.github/workflows/ci.yaml

```diff
@@ -0,0 +1,73 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: slurmdbd charm tests
+on:
+  workflow_call:
+  pull_request:
+
+jobs:
+  inclusive-naming-check:
+    name: Inclusive naming check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Run tests
+        uses: get-woke/woke-action@v0
+        with:
+          fail-on-error: true
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+
+  integration-test:
+    name: Integration tests (LXD)
+    runs-on: ubuntu-latest
+    needs:
+      - inclusive-naming-check
+      - lint
+      - unit-test
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: lxd
+          # Juju channel should eventually be updated to 3.0/stable
+          juju-channel: 2.9/stable
```

Comment:
Are are using juju 3.0 in https://github.com/canonical/slurmdbd-operator/blob/main/tox.ini#L60 

Should not use the same version on both?

Reply:
Ah! Might not seem like it, but the juju in https://github.com/canonical/slurmdbd-operator/blob/main/tox.ini#L60 is not the same as the juju that the `actions-operator` action sets up inside the CI pipeline.

The `actions-operator` action pulls juju proper (`sudo snap install juju --classic`) while the juju package that the tox file is referring to is the juju client library for Python: https://pypi.org/project/juju/. The Python client library wraps around juju so that you can control juju from your Python scripts. Let me know if you have any other questions!

Reply:
Hi @jamesbeedy! Do you mean having the version of Juju and version of the Python client library be equivalent? i.e. `assert 2.7.38 == 2.7.38`? If so, no, we do not need to use the same version of `python-libjuju` with juju. `python-libjuju` is just a client for interacting with Juju. Certain methods in `python-libjuju` just make subprocess calls to the juju CLI executable.

Here are the requirements for `python-libjuju` (known as `juju` in PyPI):

* Python 3.6+
* Juju 2.0+

Let me know if that answers your question!

Reply:
@NucciTheBoss it does. Thanks!